### ISSUE TITLE
validate_header and header_helper functions would help a lot

### ISSUE THREAD

**Issue by jprybylski** [jprybylski/quartifyr#31]
Currently there are a fairly large number of required preamble matter expected at the top of a qmd document using this tool (or in _quarto.yml, same difference). It would be good to have a convenience function `validate_header` that validates that all required fields are there in the header and/or _quarto.yml (or, for example, no conflicts exist such as title field in a memo-type report), and some nice cli-formatted output that says what has been provided, what is missing, what (optional) could be added and what each does. This should be evergreen by design.

A `header_helper` function would also be good in providing an interactive tool to guide (new) users through generating all that front-matter. The result of this should be a copyable preamble block (a la reprex push to clipboard or print to console; open in viewer like reprex if it can't access keyboard would be good if it's easy). This should also be evergreen by design.

Naturally these would require clean and thorough documentation, which is made readily discoverable given the target audience.
